### DISCUSSION TITLE
Call to undefined method Rappasoft\LaravelLivewireTables\Views\Column::components()

### DISCUSSION THREAD

**Discussion by LfJohnVo** [rappasoft/laravel-livewire-tables#511] [Q&A]
I try to create edit and delete button but i get this error

![image](https://user-images.githubusercontent.com/32848900/137669244-96567c24-a624-4b70-af36-d4744e6e3f5b.png)

somebody know hot to fix?

`<?php

namespace App\Http\Livewire;

use Illuminate\Database\Eloquent\Builder;
use Rappasoft\LaravelLivewireTables\DataTableComponent;
use Rappasoft\LaravelLivewireTables\Views\Column;
use Rappasoft\LaravelLivewireTables\Views\Link;
use App\Models\Product;

class ProductTable extends DataTableComponent
{

    public function columns(): array
    {
        return [
            Column::make('Estado', 'subcategory.name')
                ->sortable()
                ->searchable(),
            Column::make('Nombre', 'name')
                ->sortable()
                ->searchable(),
            Column::make('Precio', 'price')
                ->sortable()
                ->searchable(),
            Column::make('Estado', 'status')
                ->sortable()
                ->searchable()
                ->addClass('px-2 py-1 font-semibold leading-tight text-orange-700 bg-gray-100 rounded-sm'),
            Column::make('Actions')
                ->components([
                    Link::make('Edit') // Optionally pass false to hide the text
                        ->icon('fas fa-pencil-alt')
                        ->class('btn btn-primary btn-sm')
                        ->href(function ($model) {
                            return route('admin.auth.role.edit', $model->id);
                        }),
                ]),
        ];
    }

    public function query(): Builder
    {
        return Product::with('subcategory');
    }
}
`

**Comment by rappasoft** [comment]
The code you are using is very old, those classes don't exist anymore in this version.

https://rappasoft.github.io/laravel-livewire-tables-docs/

**Reply by LfJohnVo** [reply]
How i can create that buttons?

**Comment by rappasoft** [comment]
Just return a view from the format() method: https://rappasoft.github.io/laravel-livewire-tables-docs/columns/Using-the-built-in-cell-formatting
